Admin view

HedicGuibert opened the pull request:
This PR fixes [#85](https://github.com/jolicode/starfleet/issues/85#issue-750847364) .

It adds createdAt and speakers columns to the admin talk view. It also adds a talks column to the user list view, but this may be unnecessary.

Talks list view :
![Talks](https://user-images.githubusercontent.com/71645693/100884911-56cc8880-34b2-11eb-97af-4c84d7e11f34.png)

Users list view :
EDIT :  removed

![Users](https://user-images.githubusercontent.com/71645693/100884962-64820e00-34b2-11eb-8a8d-fb32e883ad2c.png)

lyrixx commented:
Please, add some screenshot. Thanks

HedicGuibert commented:
Sure, sorry for this omission.
I updated my innitial comment.

welcoMattic commented:
I'm afraid that the talks column on the user list view is not necessary. In fact, the talk list of an user will increase over time, so the content of this column to, it will quickly become unreadable.

lyrixx commented on src/DataFixtures/TalkFixtures.php:
```diff
@@ -0,0 +1,25 @@
+{
+    public function load(ObjectManager $manager)
+    {
+
```
useless space

HedicGuibert commented:
Ok, this makes sense, I'm removing it then

lyrixx commented on src/Entity/Talk.php:
```diff
@@ -46,9 +46,15 @@ class Talk
      */
     private $submits;
 
+    /**
+     * @ORM\OneToMany(targetEntity=User::class, mappedBy="talks")
+     */
+    private $authors;
```
Why did you add this new property? more over this seems wrong (never updated, never used)

HedicGuibert replied:
Isn't a relation needed between Talk and User to allow the speakers to be displayed on the talks list view ?

welcoMattic commented on src/Entity/Talk.php:
```diff
@@ -46,9 +46,15 @@ class Talk
+    /**
+     * @ORM\OneToMany(targetEntity=User::class, mappedBy="talks")
+     */
+    private $authors;
```
Hmm, I think there is a design issue here. Actually the data schema is:
User 1->n Submit n->1 Talk
Here, you are adding a direct relation between User and Talk, but this relation exists, indirectly through Submit. 
So, you might be able to retrieve all talks of a user through there submits, with something like:

```
    public function getAuthors(): string
    {
        return array_reduce($this->getSubmits(), function ($r, Submit $submit) {
            foreach ($submit->getUsers() as $user) {
                $r = '' === $r ? $user->getName() : $r . ', ' . $user->getName();
            }

            return $r;
        }, '');
    }
```

(I did not test this code)

HedicGuibert replied:
Oh sure, it looks a lot better indeed and adresses what is said above.
Thanks !

lyrixx replied:
And BTW, the code here is ugly. Use a template in easy admin for that

lyrixx commented on templates/easy_admin/Talk/authors.html.twig:
```diff
@@ -0,0 +1,5 @@
+{% for submit in value %}
+    {% for user in submit.users %}
```
I would use `<ul><li>` here ?

lyrixx commented on config/packages/easy_admin/talk.yaml:
```diff
@@ -7,6 +7,8 @@ easy_admin:
                     - { property: 'title', type: 'text' }
                     - { property: 'submits' }
+                    - { property: 'createdAt', type: 'date' }
+                    - { property: 'authors', type: 'string', template: 'easy_admin/Talk/authors.html.twig' }
```
`Talk` -> `talk`

lyrixx commented on src/Entity/Talk.php:
```diff
@@ -101,6 +101,11 @@ public function removeSubmit(Submit $submit): self
         return $this;
     }
 
+    public function getAuthors()
+    {
+        return $this->getSubmits();
```
The wording seems odd Author VS Submits, and anyway, I don't think this method is useful anyway :)

lyrixx commented:
Could you also sqash your commits, because it does not say what they do

lyrixx commented on config/packages/easy_admin/talk.yaml:
```diff
@@ -7,6 +7,8 @@ easy_admin:
                     - { property: 'title', type: 'text' }
                     - { property: 'submits' }
+                    - { property: 'createdAt', type: 'date' }
+                    - { property: 'authors', type: 'text', template: 'easy_admin/talk/authors.html.twig' }
```
The type here is useless since you are using a template

lyrixx commented on templates/easy_admin/talk/authors.html.twig:
```diff
@@ -0,0 +1,7 @@
+{% for submit in item.submits %}
+    <ul style="margin-bottom:0">
```
I would put the `ul` before the first for, or you'll get many ul in this table cell.
And IMHO, we will have lot's of duplicate content here.
If a user submit 5 talks. He'll appear here 5 times.
You should unique the list

welcoMattic replied:
Even if most of time there will be 1 speaker, if a talk is submitted 2 times with 2 different speakers couple (let say Alice and Bob, and Alice and Jack), we will see "Alice, Bob, Jack", which is not very relevant, don't you think? 
May be unique the list, but "by submit"
1st submit: Alice
2nd submit: Alice
3rd submit: Alice and Bob
4th submit: Alice and Jack
5th submit: Alice and Bob
will result to:
Alice, Alice + Bob, Alice + Jack

WDYT?

HedicGuibert commented:
I updated the view accordingly to what you said, and I added some fixtures.
Here is what it looks like now :
![NewTalksView](https://user-images.githubusercontent.com/71645693/100997126-3821cc80-355a-11eb-84e3-dde96da8bdfb.png)

lyrixx reviewed (approved):
I approve, even if you did not uniq the list. That's an edge and it seems you did not manage to succeed. Not a big deal.

HedicGuibert commented:
I did it locally but I felt like it wasn't useful, seems I was wrong ? I can still add it
